app/models/task.py

```diff
@@ -1,5 +1,43 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy import ForeignKey
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from datetime import datetime
+from typing import Optional
+
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from .goal import Goal
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]] = mapped_column(nullable=True)
```

Comment:
Using `Optional[]` is all that we need to indicate that a field is nullable so we don't need to also include `mapped_column` with `nullable=True`.

```suggestion
    completed_at: Mapped[Optional[datetime]] 
```

Here's the SQLAlchemy [documentation about nullability](https://docs.sqlalchemy.org/en/20/orm/quickstart.html#declare-models)

<img width="885" alt="image" src="https://github.com/user-attachments/assets/ce679238-b311-42d8-abd3-e22bc0840b7a" />